src/ngraph/frontend/onnx_import/utils/broadcasting.cpp

```diff
@@ -17,12 +17,86 @@
 #include <numeric>
 #include <vector>
 
+#include "ngraph/op/broadcast.hpp"
+#include "ngraph/op/reshape.hpp"
+
 #include "broadcasting.hpp"
 
+static std::vector<ngraph::Shape> calculate_numpy_broadcast_shape(ngraph::Shape left_shape,
+                                                                  ngraph::Shape right_shape)
+{
+    ngraph::Shape output_shape;
+    auto rank_left = left_shape.size();
+    auto rank_right = right_shape.size();
+    auto max_rank = std::max(rank_left, rank_right);
+
+    for (auto i = 0; i < (max_rank - rank_left); ++i)
+    {
+        left_shape.insert(std::begin(left_shape), 1);
+    }
+    for (auto i = 0; i < (max_rank - rank_right); ++i)
+    {
+        right_shape.insert(std::begin(right_shape), 1);
+    }
+    for (auto index = 0; index < max_rank; ++index)
+    {
+        output_shape.push_back(std::max(left_shape.at(index), right_shape.at(index)));
+    }
+
+    return {output_shape, left_shape, right_shape};
+}
+
 namespace ngraph
 {
     namespace onnx_import
     {
+        NodeVector binary_op_numpy_broadcasting(const std::shared_ptr<Node>& left,
+                                                const std::shared_ptr<Node>& right)
+        {
+            auto left_shape = left->get_shape();
+            auto right_shape = right->get_shape();
+            auto numpy_shapes = calculate_numpy_broadcast_shape(left_shape, right_shape);
+            auto output_shape = numpy_shapes.at(0);
+            auto left_full_shape = numpy_shapes.at(1);
+            auto right_full_shape = numpy_shapes.at(2);
+
+            std::vector<size_t> left_ones_positions;
+            std::vector<size_t> right_ones_positions;
+            Shape new_left_shape;
+            Shape new_right_shape;
+            for (auto index = 0; index < output_shape.size(); ++index)
```

Comment:
Add some comments. What are you aiming to do here. What values do you want to produce?